src/nvim/eval.c

```diff
@@ -562,6 +570,14 @@ void eval_init(void)
   set_vim_var_list(VV_ERRORS, list_alloc());
   set_vim_var_nr(VV_SEARCHFORWARD, 1L);
   set_vim_var_nr(VV_HLSEARCH, 1L);
+  set_vim_var_nr(VV_TYPE_NUMBER, 0);
+  set_vim_var_nr(VV_TYPE_STRING, 1);
+  set_vim_var_nr(VV_TYPE_FUNC,   2);
+  set_vim_var_nr(VV_TYPE_LIST,   3);
+  set_vim_var_nr(VV_TYPE_DICT,   4);
+  set_vim_var_nr(VV_TYPE_FLOAT,  5);
+  set_vim_var_nr(VV_TYPE_BOOL,   6);
+  set_vim_var_nr(VV_TYPE_NULL,   7);
```

Comment:
Vim doesn't have `t_null` and we don't want it either. Vim has `v:null` and `v:none`, we intentionally omitted `v:none`, the fact that it needs `v:t_none` is an example of how awkward it is.